tests/flytekit/unit/experimental/test_eager_workflows.py

```diff
@@ -275,3 +281,29 @@ async def eager_wf_flyte_directory() -> str:
 
     result = asyncio.run(eager_wf_flyte_directory())
     assert result == "some data"
+
+
+@mock.patch("flytekit.core.utils.load_proto_from_file")
+@mock.patch("flytekit.core.data_persistence.FileAccessProvider.get_data")
+@mock.patch("flytekit.core.data_persistence.FileAccessProvider.put_data")
+@mock.patch("flytekit.core.utils.write_proto_to_file")
+def test_eager_workflow_dispatch(*args):
+    """Test that event loop is preserved after executing eager workflow via dispatch."""
+
+    @eager
+    async def eager_wf():
+        await asyncio.sleep(0.1)
+        return
+
+    ctx = context_manager.FlyteContext.current_context()
+    with context_manager.FlyteContextManager.with_context(
+        ctx.with_execution_state(
+            ctx.execution_state.with_params(mode=context_manager.ExecutionState.Mode.TASK_EXECUTION)
+        )
+    ) as ctx:
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        asyncio.events.set_event_loop(loop)
```

Comment:
Is setting the event loop safe to do in a test?

Reply:
not sure, but perplexity seems to think so: https://www.perplexity.ai/search/is-asyncio-set-event-loop-safe-MXeyb8O3QPyddzGacsrRqA

A relevant reference is: https://github.com/pytest-dev/pytest-asyncio/issues/80

Reply:
even better, the pytest-asyncio library has an `event_loop` fixture